### ISSUE TITLE
Filter publications returned by TwG datapoint

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#5653]
## Feature Description

The TwG datapoint for `GET:publications` currently returns all publications a user has access to. According to the [design doc](https://docs.google.com/document/d/17ysrd8ya64T8rAgSriEo7MJxYyLf_nvm7nOn2KR248o/edit#heading=h.azdy1vcviqed), this should be filtered to a subset of publications that are relevant for the current site, but also for TwG as it can return other kinds of publications as well.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The TwG `GET:publications` endpoint's response handling should be adjusted so that the returned publications are filtered to only return those that satisfy _both_ requirements:
    * The publication has `paymentOptions -> thankStickers` set to `true`.
    * The publication has `publicationPredicates -> businessPredicates -> supportsSiteKit` set to `true` but only if the onboarding state is `ONBOARDING_COMPLETE`.

## Implementation Brief

In the `includes/Modules/Thank_With_Google.php`, modify the `parse_data_response` protected function to filter out publications that do not satisfy the above requirements.
- Get the `$response->getPublications()` to a variable.
- Using `array_filter` helper loop through the `$response->getPublications()` and do the following logic in the callback function:
  * If the `publication` object does not have `paymentOptions->thankStickers` set to `true`, remove the publication from the array.
  * If the `publication` object does not have `publicationPredicates->businessPredicates->supportsSiteKit` set to `true`, remove the publication from the array.
  * In other words, return the array of publications that satisfy both requirements.
- Return the filtered `$response->getPublications()` to the `parse_data_response` function.

### Test Coverage

- No new tests are to be added.

## QA Brief

- Ensure you have active [publications](https://publishercenter.google.com/publications) and have access to request the publications endpoint.
- Run the `getPublications` selector in the browser console - `googlesitekit.data.select('modules/thank-with-google').getPublications()`.
- Ensure the publications array of objects met the criteria mentioned in the AC.

## Changelog entry

* Ensure Thank with Google API results are filtered properly.

**Comment by techanvil** [comment]
IB :white_check_mark:

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- The TwG GET:publications endpoint's response handling are adjusted so that the returned publications are filtered to only return those that satisfy both requirements:
   - The publication has paymentOptions -> thankStickers set to true.
   - The publication has publicationPredicates -> businessPredicates -> supportsSiteKit set to true.

![image](https://user-images.githubusercontent.com/73545194/185050574-80864b32-f233-440e-a81a-8f17f306303f.png)

**Comment by aaemnnosttv** [comment]
⚠️ Looks like `publicationPredicates` and/or `businessPredicates` are only present on publications which have completed the onboarding process. I'm confirming this with the team, but if so, we'd need to remove these from the filtering otherwise we won't be able know when a user has a publication which has remaining uncompleted onboarding steps. `thankStickers` however are not affected by this so that can remain.

**Comment by aaemnnosttv** [comment]
The TwG team confirmed the above, so unfortunately we cannot filter publications to those that have `publicationPredicates->businessPredicates->supportsSiteKit` as this will not be present until the onboarding is completed (i.e. publication is fully active).

**Comment by aaemnnosttv** [comment]
I've updated the ACs and created a follow-up PR for the changes.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

The TwG GET:publications endpoint's response handling is adjusted so that the returned publications are filtered to only return those that satisfy both requirements:
  - The publication has paymentOptions -> thankStickers set to true.
  - The publication has publicationPredicates -> businessPredicates -> supportsSiteKit set to true but only if the onboarding state is ONBOARDING_COMPLETE.

![image](https://user-images.githubusercontent.com/73545194/185534404-084a4385-0a28-4a74-9687-f3bce6ebe137.png)

I tested this when the status was NONE on the `Thank with Google publication status` in the tester plugin and when I pasted the code in the QAB, the result was `undefined`.  Therefore the endpoints only appear when onboarding is completed. 

![image](https://user-images.githubusercontent.com/73545194/185534419-48dfb11b-2a01-4452-9244-301e5960dd7e.png)

**Comment by hussain-t** [comment]
Thanks for clarifying and addressing this, @aaemnnosttv 👍

**Comment by eldeloro75017** [comment]
> Thanks for clarifying and addressing this, @aaemnnosttv 👍
